main/resource-access.bs

```diff
@@ -5,28 +5,32 @@ Issue: Write introduction to the Authenticated Resource Access section.
 
 ## Hypertext Transfer Protocol ## {#http}
 
-Issue: Write Hypertext Transfer Protocol section.
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Message Syntax and Routing [[!RFC7230]]
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Semantics and Content [[!RFC7231]].
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Authentication [[!RFC7235]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Conditional Requests [[!RFC7232]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Range Requests [[!RFC7233]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Caching [[!RFC7234]].
-
-Draft:
-A Solid data pod MAY implement HTTP/2 [[!RFC7540]].
+A [=data pod=] MUST be an HTTP/1.1 server [[!RFC7230]][[!RFC7231]].
+It SHOULD additionally implement HTTP/2 [[!RFC7540]]
+to improve performance,
+especially in cases where individual clients
+are expected to send high numbers of successive requests.
+
+A data pod SHOULD use the `https` URI scheme
+in order to secure the connection between clients and servers.
+When both `http` and `https` are supported,
+all `http` URIs MUST redirect to their `https` counterparts
+using a response with a `301` status code and a `Location` header.
+
+A data pod MUST implement HTTP/1.1 Conditional Requests [[!RFC7232]]
+to ensure that updates requested by clients
+will only be applied if given preconditions are met.
+It SHOULD additionally implement HTTP/1.1 Caching [[!RFC7234]]
+to reduce bandwidth.
+A data pod MAY implement HTTP/1.1 Range Requests [[!RFC7233]]
+on large representations
+to achieve further bandwidth reduction.
+
+A data pod MUST implement HTTP/1.1 Authentication [[!RFC7235]].
```

Comment:
I wonder if it's worth being more specific here, re RFC7234 HTTP Authentication. Both in terms of - it's such a general purpose RFC, so it might make sense to specify which parts of it we care about. I'm also concerned that people might mis-interpret the mention of it as an endorsement of Basic or Digest auth schemes?

Reply:
My intention was to tackle these in the specific authentication sections (WebID-TLS and WebID-OIDC).

It's not RFC2617, so basic and digest are not included.

Reply:
But we might want to not make proxy auth mandatory though.

Reply:
Got it, makes sense.
I think my only hesitation, then, is just that, as an implementer reading that spec, it's not super clear as to what it would _mean_ for the pod to implement RFC7235? As in, which headers, what responses to expect, etc?

Reply:
Yes, I should probably state that it should conform to the server part of that spec (and for other normative HTTP refs too). This then comes down hete to understanding one response header and being able to send one request header and one status code.

Reply:
Clarified in ed5c3fbe98332ea9ab7b63569871c789108a7c57. The explicit mention of `401` leads to `WWW-Authenticate` being a MUST there.